src/v/model/topic_revision.cc

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "model/topic_revision.h"
+
+#include "absl/strings/str_split.h"
+
+namespace model {
+
+tpr tpr::from(std::string_view s) {
+    std::vector<ss::sstring> ss = absl::StrSplit(s, "_");
```

Comment:
topic names can have `_` in them. Consider using `@` or you'll have to manually get the last segment here.